src/viewport.cpp

```diff
@@ -1036,6 +1038,33 @@ static TileHighlightType GetTileHighlightType(TileIndex t)
 		}
 	}
 
+	if (_current_company < MAX_COMPANIES && _infrastructure_window_open) {
+		/* Edge case of company owned tramway on non-company owned tile */
+		if (IsTileType(t, MP_ROAD) && !IsRoadDepot(t) && HasTileRoadType(t, RTT_TRAM)) {
+			if (IsRoadOwner(t, RTT_TRAM, _current_company)) {
+				return THT_WHITE;
+			}
+		}
+
+		switch (GetTileType(t)) {
+			case MP_ROAD:
+			case MP_RAILWAY:
+			case MP_TUNNELBRIDGE:
+			case MP_WATER:
+			{
```

Comment:
We don't use braces for switch cases unless a variable is defined within the case.